src/CosmosEditorManager.ts

```diff
@@ -88,24 +88,28 @@ export class CosmosEditorManager {
     private async loadPersistedEditor(documentUri: vscode.Uri, tree: AzureTreeDataProvider): Promise<string> {
         const persistedEditors = this._globalState.get(this._persistedEditorsKey);
         //Based on the documentUri, split just the appropriate key's value on '/'
-        const editorFilePath = Object.keys(persistedEditors).find((label) => path.relative(documentUri.fsPath, label) === '');
-        if (editorFilePath) {
-            const editorNode: IAzureNode | undefined = await tree.findNode(persistedEditors[editorFilePath]);
-            let editor: ICosmosEditor;
-            if (editorNode) {
-                if (editorNode.treeItem instanceof MongoCollectionTreeItem) {
-                    editor = new MongoCollectionNodeEditor(<IAzureParentNode<MongoCollectionTreeItem>>editorNode);
-                } else if (editorNode.treeItem instanceof DocDBDocumentTreeItem) {
-                    editor = new DocDBDocumentNodeEditor(<IAzureNode<DocDBDocumentTreeItem>>editorNode);
-                } else if (editorNode.treeItem instanceof MongoDocumentTreeItem) {
-                    editor = new MongoDocumentNodeEditor(<IAzureNode<MongoDocumentTreeItem>>editorNode);
```

Comment:
The diff is actually much smaller. Just inserting an if check in line 91, and else block at 109.

Reply:
View it this way: https://github.com/Microsoft/vscode-cosmosdb/pull/436/files?w=1